src/utils.rs

```diff
@@ -138,8 +139,7 @@ where
     L: Into<usize>,
 {
     let (s, l) = (size.into(), limit.into());
-    trace!("Out of bounds: ensures {} < {}", s, l);
-    Some(()).filter(|_| s <= l).ok_or(OutOfBounds)
+    (s <= l).then_some(()).ok_or(OutOfBounds)
```

Comment:
Why remove the log? then_some is awesome btw!

Reply:
Yeah i should stop doing that, we can revert that line.